pulp_container/app/utils.py

```diff
@@ -144,9 +144,14 @@ def has_task_completed(dispatched_task):
 
 def determine_media_type(content_data, response):
     """Determine the media type of a manifest either from the JSON data or the response object."""
-    if media_type := content_data.get("mediaType"):
+    if media_type := response.headers.get("content-type"):
```

Comment:
Can you keep the logic as it was before? I.e the ultimate source of truth is `content_data.get("mediaType")` if available, then `response.headers.get("content-type")` and as last you continue parsing the json file.